### DISCUSSION TITLE
Having trouble installing the package in HardHat

### DISCUSSION THREAD

**Discussion by kkoshiya** [PaulRBerg/prb-math#208] [Q&A] [answered]
Hey community, I'm currently trying to get started using the prb-math package for my contracts via hardhat (I can't use foundry due to a different conflict). I was wondering if somebody could help me, for some reason my contract can't find the import. Thanks

**Comment by PaulRBerg** [comment]
hey @kkoshiya, thanks for your question, and sorry to hear that you are encountering this difficulty.

Would you be able to add more detail about what exactly went wrong? Ideally, if you could reproduce the error in an open-source repo, that would be perfect.

**Reply by kkoshiya** [reply]
Thanks for the response Paul! Yeah so basically I tried to follow the instructions to install @prb/math into my packages file. I'm not sure if it's correctly loading into the node-mods. However, when I run npx hardhat compile, I get the error that "File @prb/math/SD59x18.sol, imported from contracts/test.sol, not found." If you could please help me figure out what is going on here, that would be very much appreciated. Here is the link to the repo: https://github.com/kkoshiya/CauchyAMM

**Reply by PaulRBerg** [reply]
Thanks @kkoshiya, see my answer below.

**Comment by PaulRBerg** [comment, marked as answer]
The instructions in the README pertain to a Foundry environment in which the PRBMath submodules are remapped to remove the `src`. But in Hardhat and Node.js, that is not possible, so you have to import the contracts using the full paths.

As you can see in the node modules installed in [CauchyAMM](https://github.com/kkoshiya/CauchyAMM), the `@prb/math` path contains a `src` directory:

https://github.com/kkoshiya/CauchyAMM/tree/main/node_modules/%40prb/math

So the solution is to change this line:

```solidity
import { SD59x18 } from "@prb/math/SD59x18.sol";
```

To this:

```solidity
import { SD59x18 } from "@prb/math/src/SD59x18.sol";
```

Side note: I will update the imports in the README to use `src` because now I recommend this approach even for Foundry projects.

**Reply by kkoshiya** [reply]
Thanks Paul, it worked! my contract compiles w/ the import. Sorry for the dumb question, I prob should've noticed the path issue myself. Very much appreciated!

**Reply by PaulRBerg** [reply]
No worries at all - I love answering questions like this.

Good luck 🍀
